lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -58,9 +91,25 @@ def max_login_attempts
           self.class.max_login_attempts
         end
 
-        def populate_otp_column
+        def totp_enabled?
+          self.respond_to?(:otp_secret_key) && !self.otp_secret_key.nil?
+        end
+
+        def create_otp_secret
           self.otp_secret_key = ROTP::Base32.random_base32
         end
+
+        def create_direct_otp(options = {})
+          # Create a new random OTP and store it in the database
+          digits = options[:direct_otp_length] || self.class.direct_otp_length
+          update_columns(direct_otp: random_base10(digits), direct_otp_sent_at: Time.now.utc)
```

Comment:
But I'm converting directly to UTC, so shouldn't Time.now.utc so it doesn't matter here does it?  System time should be fine.